Solutions/FortiWebCloud/Workbooks/Fortiweb-workbook.json

```diff
@@ -0,0 +1,367 @@
+                    "durationMs": 43200000
+                  },
+                  {
+                    "durationMs": 86400000
+                  },
+                  {
+                    "durationMs": 172800000
+                  },
+                  {
+                    "durationMs": 259200000
+                  },
+                  {
+                    "durationMs": 604800000
+                  },
+                  {
+                    "durationMs": 1209600000
+                  },
+                  {
+                    "durationMs": 2419200000
+                  },
+                  {
+                    "durationMs": 2592000000
+                  },
+                  {
+                    "durationMs": 5184000000
+                  },
+                  {
+                    "durationMs": 7776000000
+                  }
+                ],
+                "allowCustom": true
+              },
+              "timeContext": {
+                "durationMs": 86400000
+              }
+            }
+          ],
+          "style": "pills",
+          "queryType": 0,
+          "resourceType": "microsoft.operationalinsights/workspaces"
+        },
+        "name": "parameters - 1"
+      },
+      {
+        "type": 3,
+        "content": {
+          "version": "KqlItem/1.0",
+          "query": "Fortiweb\n| summarize count() by EventSubType",
+          "size": 0,
+          "title": "Attack Type",
+          "timeContextFromParameter": "TimeRange",
+          "queryType": 0,
+          "resourceType": "microsoft.operationalinsights/workspaces",
+          "visualization": "piechart"
+        },
+        "customWidth": "25",
+        "name": "Attack Type"
+      },
+      {
+        "type": 3,
+        "content": {
+          "version": "KqlItem/1.0",
+          "query": "Fortiweb\n| where isnotempty(RequestClientApplication)\n| summarize count() by RequestClientApplication ",
+          "size": 0,
+          "title": "Client Application",
+          "timeContextFromParameter": "TimeRange",
+          "queryType": 0,
+          "resourceType": "microsoft.operationalinsights/workspaces",
+          "visualization": "piechart"
+        },
+        "customWidth": "25",
+        "name": "Client Application"
+      },
+      {
+        "type": 3,
+        "content": {
+          "version": "KqlItem/1.0",
+          "query": "Fortiweb\n| where isnotempty(RequestMethod)\n| summarize count() by RequestMethod ",
+          "size": 0,
+          "title": "Request Methods",
+          "timeContextFromParameter": "TimeRange",
+          "queryType": 0,
+          "resourceType": "microsoft.operationalinsights/workspaces",
+          "visualization": "piechart"
+        },
+        "customWidth": "25",
+        "name": "Request Methods"
+      },
+      {
+        "type": 3,
+        "content": {
+          "version": "KqlItem/1.0",
+          "query": "Fortiweb\n| where isnotempty(owaspTop10)\n| summarize count() by owaspTop10 ",
```

Comment:
These fields should either have columnifexist or if we are sure that we fetch data from them from specific custom string, we can extend the same in parser.
If neither of this is true, when data is not present, query will fail.

Reply:
Parser modified